### DISCUSSION TITLE
MQTT Configuration by API?

### DISCUSSION THREAD

**Discussion by hacki11** [fredlcore/BSB-LAN#265] [Q&A] [answered]
I have no complete overview of the new features and couldn't find it in the manual, yet.

Is it possible to configure the "wanted" mqtt-parameter by an API?
I found it is configured by compile time, but if it is possible to configure this by a (restful) web api that would benefit the iobroker users as follow:

Currently i'm polling the values by a user given interval. But if configuration and reception over mqtt can be done by API, i can offer the users more "live" values with pushing instead of polling.

**Comment by 1coderookie** [comment]
> I have no complete overview of the new features and couldn't find it in the manual, yet.

Yes I know, sorry, but right now there are _so many_ changes within different functions, that it's kinda hard for me to keep it all up-to-date in 'realtime'. Besides that, not all changes and new functions are always clear to me and I have to ask @fredlcore or @dukess for specific and further informations. But I'll try my best to stay on track.. ;)

**Comment by fredlcore** [comment, marked as answer]
MQTT parameters are the same as log parameters, so they can be set via the /L URL command and queried via /JI.

**Comment by torbennehmer** [comment]
> Is it possible to configure the "wanted" mqtt-parameter by an API?

right now this is not possible as far as I understand the codebase. At this point I am toying around with an MQTT topic structure, that would allow that in #346, where this would be covered by MQTT messages.
